Add provider test for adding features with missing attributes

nyalldawson opened the pull request:
Add provider test to ensure that features added with missing attributes
are transparently padded out with NULL attributes to the required fields
length

Currently the behavior is inconsistent - some providers reject these
features, others pad them out, and worse -- some add them with
missing attributes (memory provider), causing ALL sorts of flow-on,
difficult to debug issues.

nyalldawson committed:
Add provider test to ensure that features added with missing attributes
are transparently padded out with NULL attributes to the required fields
length

Currently the behavior is inconsistent - some providers reject these
features, others pad them out, and worse -- some add them with
missing attributes (memory provider), causing ALL sorts of flow-on,
difficult to debug issues.

nyalldawson commented:
Alrighty... let's see which providers are badly behaved here....

nirvn commented:
```
The following tests FAILED:
	274 - PyQgsMemoryProvider (Failed)
	313 - PyQgsShapefileProvider (Failed)
	364 - PyQgsPostgresProvider (Failed)
```

nyalldawson committed:
Fix memory provider handling of missing attributes when adding features

Ensure that features added to a memory provider (and returned when
fetching features from a memory provider) always have the correct
number of attributes present.

Fixes many random behavior bugs when working with memory providers.
